"+" UIBarButton item  to add list

pravalgautam opened the pull request:
Implements a "+" from #4188

pravalgautam commented:
![Simulator Screen Shot - iPhone 14 Pro - 2023-02-15 at 11 43 50](https://user-images.githubusercontent.com/71179696/218999374-93012e54-ce1f-4ee9-8a0e-0ba7e7550977.png)

biodranik commented:
Please create separate PRs for separate fixes/features.

biodranik commented:
The bottom button "Create a new list" can also stay in addition to a + button. On large modern devices, it's more convenient to tap a button at the bottom.

The plan is also to add the "Export all bookmarks" button at the bottom, so both buttons can sit together nicely.

pravalgautam commented:
> The bottom button "Create a new list" can also stay in addition to a + button. On large modern devices, it's more convenient to tap a button at the bottom.
> 
> The plan is also to add the "Export all bookmarks" button at the bottom, so both buttons can sit together nicely.

![Simulator Screen Shot - iPhone 14 Pro - 2023-02-15 at 16 05 27](https://user-images.githubusercontent.com/71179696/219003833-72e0d1cd-5b7f-4455-a02c-138b75327766.png)

biodranik commented on iphone/Maps/UI/PlacePage/Components/PlacePagePreviewViewController.swift:
```diff
@@ -202,7 +202,7 @@ final class PlacePagePreviewViewController: UIViewController {
       }
       
-      setScheduleLabel(state: L("closed_now"),
+      setScheduleLabel(state: L(""),
```
Is it from another PR?

pravalgautam replied:
yes

pravalgautam replied:
this  is the PR belong to this isssue https://github.com/organicmaps/organicmaps/pull/4491/commits/cbfc515a8c821f5042f8341dc5eb2b10ef095db5

pravalgautam replied:
this PR is cbfc515a8c821f5042f8341dc5eb2b10ef095db5

vng reviewed (commented):
@pravalgautam @biodranik 
Next steps here? LGTM

biodranik commented on xcode/shaders/shaders.xcodeproj/project.pbxproj:
```diff
@@ -176,7 +176,7 @@
 		45789EE52135464D009955CC /* metal_program_params.hpp */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.cpp.h; path = metal_program_params.hpp; sourceTree = "<group>"; };
 		45789EE62135464D009955CC /* metal_program_params.mm */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.cpp.objcpp; path = metal_program_params.mm; sourceTree = "<group>"; };
-		4598437C21394BE000F8CAB2 /* shaders_metal.metallib */ = {isa = PBXFileReference; includeInIndex = 0; lastKnownFileType = "archive.metal-library"; path = shaders_metal.metallib; sourceTree = BUILT_PRODUCTS_DIR; };
+		4598437C21394BE000F8CAB2 /* shaders_metal.metallib */ = {isa = PBXFileReference; explicitFileType = "archive.metal-library"; includeInIndex = 0; path = shaders_metal.metallib; sourceTree = BUILT_PRODUCTS_DIR; };
```
Why is this change necessary?

biodranik commented on iphone/Maps/UI/PlacePage/Components/PlacePagePreviewViewController.swift:
```diff
@@ -1,242 +0,0 @@
-final class PlacePageDirectionView: UIView {
```
Why is this file deleted?

biodranik commented on iphone/Maps/Bookmarks/Categories/BMCViewModel/BMCDefaultViewModel.swift:
```diff
@@ -117,8 +117,8 @@ extension BMCDefaultViewModel {
 
-    categories.append(manager.category(withId: manager.createCategory(withName: name)))
-    view?.insert(at: [IndexPath(row: categories.count - 1, section: section)])
+    categories.insert(manager.category(withId: manager.createCategory(withName: name)), at: 0)
```
Why is this change necessary?

biodranik replied:
As I understood from your testing, there are some issues with this approach, right?

Let's remove these lines from this PR so it can be merged. And implement sorting in another PR.

pravalgautam commented:
> Can you please write "Closes #issuenumber" in the PR description, to reference the issue that you're fixing?

 I am sorry for the clutter I have made 
the commit that I fixed f874c2ed28495b3fd484bb1166173bfd2080d62c
I fixed the  and als I am sorry for the clutter I have made 
the commit that I fixed f874c2ed28495b3fd484bb1166173bfd2080d62c
mentioned issuenumber  the and signed (DCO) also

pravalgautam commented:
> Can you please write "Closes #issuenumber" in the PR description, to reference the issue that you're fixing?

 I am sorry for the clutter that  I have made
solved issue #4188 commit --https://github.com/organicmaps/organicmaps/commit/f874c2ed28495b3fd484bb1166173bfd2080d62c
mentioned issuenumber the and signed (DCO) also

biodranik commented on iphone/Maps/Bookmarks/Categories/BMCView/BMCViewController.swift:
```diff
@@ -35,6 +35,12 @@ final class BMCViewController: MWMViewController {
     super.viewDidLoad()
     viewModel = BMCDefaultViewModel()
+    
+    let addButton = UIBarButtonItem(barButtonSystemItem: .add, target: self, action: #selector(addTapped))
+    navigationItem.rightBarButtonItem = addButton
+  }
+    @objc func addTapped() {
+    createNewCategory()
```
Will this work? (you likely need to add `@objc` annotation to createNewCategory() method.

```suggestion
    navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .add, target: self, action: #selector(createNewCategory))
```

biodranik replied:
Did you test it?

biodranik reviewed (commented):
I've updated the PR description.

Did you test list deletion and OM restarting? Does the order of lists stay the same?

pravalgautam commented:
> I've updated the PR description.
> 
> Did you test list deletion and OM restarting? Does the order of lists stay the same?

I need to fix this after relaunching all the lists are deleted

pravalgautam commented:
> I've updated the PR description.
> 
> Did you test list deletion and OM restarting? Does the order of lists stay the same?

yes I test deletion and the lists stay the same  but Resolved -- if we add  lists and directly discard the app(using swipe up ) without using the navigation button then the lists disappeared too
the commit for this---11bab0224195d04170979f006146c0f8b448a5ba

biodranik reviewed (changes requested):
The PR looks messed up. Did you learn about and try to git rebase and git commit --amend?

Let's split this issue into two. Please leave only adding of a + button in this PR. And create a new PR for the sorting issue.

pravalgautam commented:
> The PR looks messed up. Did you learn about and try to git rebase and git commit --amend?
> 
> Let's split this issue into two. Please leave only adding of a + button in this PR. And create a new PR for the sorting issue.

Yes, I learned about the rebase but sir do I have to make an issue  and then pull a request a PR to it

biodranik commented:
Do you see open unresolved comments?

pravalgautam commented:
> Do you see open unresolved comments?

yes
